RFC: DCP Incident Response Plan

stahiri opened the pull request:
The DCP needs to have a confirmed plan and procedures in place so that we are prepared to respond to an incident. This RFC proposes an Incident Response Plan for review.

~~**June 14**: Last call for community review~~
~~**June 21**: Last call for community review (extended 1 week)~~
~~**July 12**: Last call for Oversight review~~
Review closed; pending final approval.

stahiri committed:
Create 0000-Incident-Response-Plan.md

brianraymor commented on rfcs/text/0000-Incident-Response-Plan.md:
```diff
@@ -0,0 +1,52 @@
+
+## Detailed Design
+
+Please review the draft [DCP Incident Response Plan](https://docs.google.com/document/d/1DW-WskpfEQHaFhnj4RFMAB5rf2W5OCpeUb19uXC2npI) which, at a high level, covers the following:
```
That google doc *must* be incorporated into this RFC for review to continue. I don't see how an RFC can be approved if the details are in a separate document that can be quietly versioned. We do not want the precedent of using a RFC as a "frame" for information that's documented elsewhere. In addition, the document in question is a *draft* which requires the community to review two documents. We want the history of comments to be captured in one place.

stahiri replied:
Thanks for this feedback - the file has been updated to reflect the text of the IRP Google Doc.

brianraymor commented on rfcs/text/0000-Incident-Response-Plan.md:
```diff
@@ -0,0 +1,52 @@
+
+### Acceptance Criteria
+
+The IRP must be confirmed as executable by all teams within the DCP community.
```
This would seem to be the purpose of the community and oversight review? Perhaps the A/C could be that a real incident is correctly managed by the process; otherwise, it would be updated as required?

stahiri replied:
Acceptance Criteria has been updated in the updated file

brianraymor reviewed (changes requested):
This RFC is not ready for review because it references an external draft google doc for its detailed design.

stahiri committed:
Update 0000-Incident-Response-Plan.md

Embed IRP content into 0000-Incident-Response-Plan.md

tjchen commented on rfcs/text/0000-Incident-Response-Plan.md:
```diff
@@ -30,23 +30,169 @@ As DCP governance, I want be notified of a confirmed or suspected incident so th
+1. If known, also include:
+   1. Date incident occurred
+   1. Date incident was discovered
+   1. Affected system(s) and application(s)
```
including version?

stahiri replied:
yes - will add this to the description

stahiri replied:
Added!

tjchen commented on rfcs/text/0000-Incident-Response-Plan.md:
```diff
@@ -30,23 +30,169 @@ As DCP governance, I want be notified of a confirmed or suspected incident so th
+1. Date of discovery of the suspected or confirmed incident
+1. Brief description of suspected or confirmed incident
+1. If known, also include:
+   1. Date incident occurred
```
I have seen before where the reporter can classify the ticket initially (event, incident, etc) and then the reviewer can downgrade/change as needed. In this situation, are all defaulting to a certain severity?

stahiri replied:
I've updated to specifically called this out within today's updates

tjchen commented on rfcs/text/0000-Incident-Response-Plan.md:
```diff
@@ -30,23 +30,169 @@ As DCP governance, I want be notified of a confirmed or suspected incident so th
+* __Step 3.1__: If the DCP IRT determines that the event does not constitute a security incident or is a false positive, the DCP IRT Coordinator will document the determination and supporting evidence within the ticket and close the ticket. No further action is required.
+* __Step 3.2__: If the DCP IRT confirms that the event does constitute a security incident, the DCP IRT Coordinator will collaborate with the Incident Handler to save all evidence and complete an Incident Report (see Appendix B).
+
+__Step 4__: The DCP IRT Coordinator will update the incident’s ticket to include a link to the completed Incident Report and notify the following that the report has been completed:
```
If it is something that requires a postmortem, the report-out could be valuable for the entire DCP (or at least stakeholders with relevant components)

stahiri replied:
Sharing out to the rest of DCP after the incident response process is closed is perfectly acceptable.

stahiri committed:
Update 0000-Incident-Response-Plan.md

Updates based on TJ's comments: (1) request affected system and application versions + estimated incident severity rating during incident ticket creation and (2) allow IRT the opportunity to revise the incident severity rating after incident review

Bento007 commented on rfcs/text/0000-Incident-Response-Plan.md:
```diff
@@ -0,0 +1,199 @@
+
+Security incident response has become increasingly important as cybersecurity attacks have become more frequent, diverse, and damaging. Though preventive security measures can reduce the frequency of incidents, they cannot stop them all in their tracks. It is therefore necessary for DCP to have a documented Incident Response Plan that guides us in identifying, investigating, containing, mitigatin ...
+
+Additionally, we want the DCP system to be secure and reliable for all stakeholders, including engineering teams, program governance, and, above all, end users. In order to assure that we are following security best practices, and to help us attest to that fact in the future, we are working toward compliance with the NIST SP 800-53 Rev 4 security framework. Incident response is one of the require ...
```
add a link to `NIST SP 800-53 Rev 4 security framework` for reference.

stahiri replied:
Will be linked in next update

austinylin commented on rfcs/text/0000-Incident-Response-Plan.md:
```diff
@@ -0,0 +1,199 @@
+#### Incident Response Procedures
+Anyone can declare an incident. Follow the procedure outlined below if you suspect or have confirmed that an incident has occurred.
+
+__Step 1__: Personnel identifying the suspected or confirmed incident (Incident Handler) will alert the DCP IRT by submitting a ticket to the to-be-created humancellatlas/dcp-security private GitHub repo. The ticket will include the following information to ensure appropriate triage:
```
Will this trigger PagerDuty or another alerting tool for high severity incidents?

stahiri replied:
No; this IRP doesn't include Security Operations Center (SOC) level response. We may one day require that, but it will be a future enhancement to process as needed.

austinylin commented on rfcs/text/0000-Incident-Response-Plan.md:
```diff
@@ -0,0 +1,199 @@
+### Roles and Responsibilities
+The primary individuals and roles listed below play a part in the DCP incident reporting and response processes.
+* __DCP Incident Response Team__: The DCP IRT serves as the focal point for IT security incidents for the DCP. The DCP IRT identifies security incidents, characterizes the nature and severity of incidents, and provides diagnostic and corrective actions when appropriate. The IRT will receive any reports of suspicious or confirmed incidents. The DCP IRT will be comprised of DCP Security Leads and s ...
+* __Incident Response Team Coordinator__: The IRT Coordinator is a member of the IRT responsible for assembling all data pertinent to an incident, communicating with the appropriate parties, ensuring information is complete, overseeing the handling of the incident, and reporting on incident status both during and after investigation. This individual ensures the procedures outlined in this plan ar ...
```
Is there a separate document that outlines guidance for when the IRT Coordinator will notify folks like legal, communications, etc? Personally, I've found having that written down for folks leading response to be very helpful, especially when incidents aren't very frequent.

stahiri replied:
There are high-level communication/escalation steps outlined in the process portion of this IRP. Per this plan, the IRT Coordinator is responsible for notifying DCP security, business stakeholders, and the Oversight Committee; legal, communications teams, etc. will be notified based on feedback from Oversight either by Oversight or as Oversight otherwise directs.

austinylin commented on rfcs/text/0000-Incident-Response-Plan.md:
```diff
@@ -0,0 +1,199 @@
+#### Incident Response Procedures
+Anyone can declare an incident. Follow the procedure outlined below if you suspect or have confirmed that an incident has occurred.
+
+__Step 1__: Personnel identifying the suspected or confirmed incident (Incident Handler) will alert the DCP IRT by submitting a ticket to the to-be-created humancellatlas/dcp-security private GitHub repo. The ticket will include the following information to ensure appropriate triage:
```
What happens if someone finds an issue but doesn't have access to that project? How do they notify?

stahiri replied:
End users - or anyone finding they don't have appropriate access to the github repo - can report things to data-help@humancellatlas.org where the reports can be triaged by the team to the repo. Will add text around this.

austinylin commented on rfcs/text/0000-Incident-Response-Plan.md:
```diff
@@ -0,0 +1,199 @@
+
+Additionally, we want the DCP system to be secure and reliable for all stakeholders, including engineering teams, program governance, and, above all, end users. In order to assure that we are following security best practices, and to help us attest to that fact in the future, we are working toward compliance with the NIST SP 800-53 Rev 4 security framework. Incident response is one of the require ...
+
+### User Stories
```
Are reports from folks outside of the DCP team in scope? (e.g. security researchers)

stahiri replied:
We should accept reports from all sources. The "how" part of this question is tbd.

austinylin commented on rfcs/text/0000-Incident-Response-Plan.md:
```diff
@@ -0,0 +1,199 @@
+
+If there is confirmation of an incident, the IRT formulates an overall response strategy and reports the incident to the Oversight Committee and other concerned parties, as appropriate.
+
+##### Phase 3: Containment, Eradication, and Recovery
```
I've found it useful in the past to explicitly call out that folks are empowered to take, at time drastic measures to close holes. For example, in the rare situation that there is evidence that data is being actively exfiltrated, shutting down the entire service might be appropriate.

stahiri replied:
As an enhancement to our IR processes, we'll develop playbooks that address specific types of security incidents. I think this information is a great candidate for inclusion in those playbooks. I'd like to keep the IRP itself quite high-level so that information within doesn't overlap with the playbooks.

stahiri committed:
Update 0000-Incident-Response-Plan.md

stahiri committed:
Update 0000-Incident-Response-Plan.md

stahiri committed:
Update 0000-Incident-Response-Plan.md

stahiri commented:
Approved during 7/25 PM meeting
